### PR TITLE
Fix #2137: Two defects in j.n.PlainSocketImpl#read

### LINKED ISSUE #2137: Two defects in j.n.PlainSocketImpl#read(buffer, offset, count)
Whilst studying Issue #2131, I noticed two defects in `java.net.PlainSocketImpl#read(buffer, offset, count)`.

* The code does not raise exceptions for non-0 errnos other than EAGAIN or EWOULDBLOCK.
Such errors are unlikely, but it would be nice to know about them when they occur.

* The code does not properly return 0 when the `count` parameter is 0.  The call should succeed
   rather than returning -1 (EOF).  Ref: https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#read-byte:A-int-int-

### PR DESCRIPTION
**Pull request by LeeTibbert**
We correct two defects in the read method for java.net.PlainSocketImpl.

### CODE REVIEW

**Review by ekrich — COMMENTED** [review]
I think this would look better if you just `throw` from the `if` and `else` branches.

**Comment by LeeTibbert** [comment]
@ekrich Thank you for the review & suggestion. Both lead to the improvement of this submission.

**Review by WojciechMazur — APPROVED** [review]

File: javalib/src/main/scala/java/net/PlainSocketImpl.scala [hunk 1]
```diff
@@ -389,13 +389,17 @@ private[net] class PlainSocketImpl extends SocketImpl {
     val bytesNum = socket
       .recv(fd.fd, buffer.asInstanceOf[ByteArray].at(offset), count.toUInt, 0)
       .toInt
-    if (bytesNum <= 0) {
-      if (errno.errno == EAGAIN || errno.errno == EWOULDBLOCK) {
+
+    bytesNum match {
+      case _ if (bytesNum > 0) => bytesNum
+
+      case _ if (bytesNum == 0) => if (count == 0) 0 else -1
```
Comment (lolgab) [inline javalib/src/main/scala/java/net/PlainSocketImpl.scala:396]:
```suggestion
      case 0 => if (count == 0) 0 else -1
```

Comment (LeeTibbert) [reply javalib/src/main/scala/java/net/PlainSocketImpl.scala:396]: @lolgab Thank you for the simplification.  I read this after this PR had been merged, so I created
PR #2149.  This code will be around for a long time and no sense letting an eyesore endure.

**Pull request merged**